File: CPP/0448FindAllNumbersDisappearedInAnArray.cpp
#include <iostream>
#include <cstdlib>
#include <vector>

using namespace std;

vector<int> findDisappearedNumber(vector<int>& nums) {
    int n = nums.size();

    for (int i = 0; i < n; i++) {
        int val = abs(nums[i]);
        int idx = val - 1;

        if (nums[idx] > 0) {
            nums[idx] = -nums[idx];
        }
    }
    
    vector<int> missing;
    for (int i = 0; i < n; i++) {
        if (nums[i] > 0) {
            missing.push_back(i + 1);
        }
    }

    return missing;
}

File: Python/0448FindAllNumbersDisappearedInAnArray.py
def findDisappearedNumbers(nums : list[int]) -> list[int]:
    n = len(nums)

    for i in range(n):
        val = abs(nums[i])
        idx = val - 1

        if nums[idx] > 0:
            nums[idx] = -nums[idx]
    
    missing = []
    for i in range(n):
        if nums[i] > 0:
            missing.append(i + 1)
    
    return missing